account_financial_report/report/general_ledger.py

```diff
@@ -459,78 +459,86 @@ def _get_period_ml_data(
         if extra_domain:
             domain += extra_domain
         ml_fields = self._get_ml_fields()
-        move_lines = self.env["account.move.line"].search_read(
-            domain=domain, fields=ml_fields, order="date,move_name"
-        )
         journal_ids = set()
         full_reconcile_ids = set()
         taxes_ids = set()
         analytic_ids = set()
         full_reconcile_data = {}
         acc_prt_account_ids = self._get_acc_prt_accounts_ids(company_id, grouped_by)
-        for move_line in move_lines:
-            journal_ids.add(move_line["journal_id"][0])
-            for tax_id in move_line["tax_ids"]:
-                taxes_ids.add(tax_id)
-            for analytic_account in move_line["analytic_distribution"] or {}:
-                analytic_ids.add(int(analytic_account))
-            if move_line["full_reconcile_id"]:
-                rec_id = move_line["full_reconcile_id"][0]
-                if rec_id not in full_reconcile_ids:
-                    full_reconcile_data.update(
-                        {
-                            rec_id: {
-                                "id": rec_id,
-                                "name": move_line["full_reconcile_id"][1],
+        aml_obj = self.env["account.move.line"]
+        batch = 1000
+        aml_ids = aml_obj.search(domain).ids
+        while aml_ids:
+            current_amls = aml_obj.search([("id", "in", aml_ids)], limit=batch)
+            move_lines = aml_obj.search_read(
```

Comment:
Why search with limit and then search_read those records. Can't you directly search read?

Reply:
Corrected in new version